orchestrator/careplancontributor/ehr/kafka_client.go

```diff
@@ -0,0 +1,169 @@
+package ehr
+
+import (
+	"fmt"
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+	"github.com/labstack/gommon/log"
+	"os"
+	"strings"
+)
+
+// KafkaConfig holds the configuration settings for connecting to a Kafka broker.
+// It includes options to enable Kafka, specify the topic, endpoint, and connection string.
+//
+// Fields:
+//   - Enabled: A boolean indicating whether Kafka is enabled.
+//   - Topic: The Kafka topic to which messages will be sent.
+//   - Endpoint: The Kafka broker endpoint.
+//   - ConnectionString: The connection string used for authentication.
+type KafkaConfig struct {
+	Enabled  bool           `koanf:"enabled" default:"false"`
+	Topic    string         `koanf:"topic"`
+	Endpoint string         `koanf:"endpoint"`
+	Sasl     SaslConfig     `koanf:"sasl"`
+	Security SecurityConfig `koanf:"security"`
+}
+
+type SaslConfig struct {
+	Mechanism string `koanf:"mechanism" default:"PLAIN"`
+	Username  string `koanf:"username" default:"$ConnectionString"`
+	Password  string `koanf:"password"`
```

Comment:
Did you check whether we have to support Azure Managed Identity as well? All components that connect to Azure services use managed Idenitity for auth